src/main/kotlin/org/elm/workspace/compiler/ElmBackAction.kt

```diff
@@ -0,0 +1,29 @@
+package org.elm.workspace.compiler
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.util.messages.Topic
+import org.elm.ide.toolwindow.ElmCompilerPanel
+
+class ElmBackAction(private val elmCompilerPanel: ElmCompilerPanel) : AnAction() {
+
+    override fun update(e: AnActionEvent) {
+        e.presentation.icon = AllIcons.Actions.Back
+        e.presentation.isEnabled = !elmCompilerPanel.compilerMessages.isEmpty()
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project
+                ?: return
+        project.messageBus.syncPublisher(ERRORS_BACK_TOPIC).back()
```

Comment:
I guess this is OK, but you may want to check how other JetBrains actions are implemented to see how they do it.

I would bet that there are a bunch of examples in the codebase where they handle "back/forward"-style actions.